generated/kbapi/README.md

```diff
@@ -0,0 +1,47 @@
+# Generated Kibana API client
+
+This package includes an API client generated from the Kibana OpenAPI spec.
+This client is very much a work in progress, there's opportunity for improvement both within this provider, but also within the OpenAPI spec created during the Kibana build process.
+
+This readme serves to document the status quo, contributions improving this current process are very much welcome.
+
+## Adding new endpoints
+
+There's more detail on the full process below, this section aims to provide a tl;dr to what's required to cover new endpoints in the generated client.
+
+1. `make transform`. This ensures you're starting with the latest API spec. If you encounter issues at this point, at least you know it's not related to any changes you've made :)
+1. Add the required API paths to the allow list. See `transformFilterPaths` in `transform_schema.go`. The definition here is straightforward, the path must match exactly what's in the API spec.
+1. If endpoints don't use any `oneOf` fields then you're likely done. Run `make transform generate` and start work on the provider resource. Sweeping statements like this tend to be wrong at times, if you encounter issues with the generated client then you'll have to investigate them and may need to define additional transforms.
+1. If the endpoints use `oneOf` in the request/response bodies then you'll need to define additional transforms to extract those to reusable components. See `transformKibanaPaths` or `transformFleetPaths` for some examples.
+1. `make transform generate` and go test the fresh client.
+
+## Client generation
+
+The actual final Go client generation is relatively straightforward. `make generate` executes [`oapi-codegen`](https://github.com/oapi-codegen/oapi-codegen) which creates `kibana.gen.go` from `oas-filtered.yaml`.
+
+## OpenAPI spec transforms.
+
+This is where the dragons lie.
+
+`make transform` downloads the schema in kibana/main, and passes it through `transform_schema.go`. This transform step aims to patch over:
+* Issues with the Kibana schema, any issues we encounter should also be raised in the Kibana repo.
+* Issues with oapi-codegen, similarly we should ensure there's a corresponding issue in the oapi-codegen repo.
```

Comment:
Does it make sense to try another tool, e.g. ogen?

Reply:
Yep, it's definitely worth investigating. I've tried a bunch of these previously (I can't remember if that includes ogen) and there bugs in every generator I tried. It's worth checking if ogen works better with the current Kibana spec though.